src/client/Item/Items.ts

```diff
@@ -72,6 +81,72 @@ export class Items {
     return new QueryIterator(this.clientContext, query, options, fetchFunction, this.container.url);
   }
 
+  /**
+   * Create a `ChangeFeedIterator` to iterate over pages of changes
+   * @param partitionKey
+   * @param changeFeedOptions
+   */
+  public readChangeFeed(
+    partitionKey: string | number | boolean,
+    changeFeedOptions: ChangeFeedOptions
+  ): ChangeFeedIterator<any>;
+  /**
+   *
+   * @param changeFeedOptions
+   */
+  public readChangeFeed(changeFeedOptions?: ChangeFeedOptions): ChangeFeedIterator<any>;
+  /**
+   *
+   * @param partitionKey
+   * @param changeFeedOptions
+   */
+  public readChangeFeed<T>(
+    partitionKey: string | number | boolean,
+    changeFeedOptions: ChangeFeedOptions
+  ): ChangeFeedIterator<T>;
+  /**
+   *
+   * @param changeFeedOptions
+   */
+  public readChangeFeed<T>(changeFeedOptions?: ChangeFeedOptions): ChangeFeedIterator<T>;
+  public readChangeFeed<T>(
+    partitionKeyOrChangeFeedOptions?: string | number | boolean | ChangeFeedOptions,
+    changeFeedOptions?: ChangeFeedOptions
+  ): ChangeFeedIterator<T> {
+    let partitionKey: string | number | boolean;
+    if (!changeFeedOptions && isChangeFeedOptions(partitionKeyOrChangeFeedOptions)) {
+      partitionKey = undefined;
+      changeFeedOptions = partitionKeyOrChangeFeedOptions;
+    } else if (partitionKeyOrChangeFeedOptions !== undefined && !isChangeFeedOptions(partitionKeyOrChangeFeedOptions)) {
+      partitionKey = partitionKeyOrChangeFeedOptions;
+    }
+
+    if (!changeFeedOptions) {
+      throw new Error("changeFeedOptions must be a valid object");
+    }
+
+    if (partitionKey !== undefined) {
+      if (changeFeedOptions.partitionKey !== undefined && partitionKey !== changeFeedOptions.partitionKey) {
+        throw new Error(
+          "Two different partition keys were specified via the partition key parameter and changeFeedOptions"
+        );
+      }
+      changeFeedOptions.partitionKey = partitionKey;
+    }
+
+    const path = Helper.getPathFromLink(this.container.url, "docs");
+    const id = Helper.getIdFromLink(this.container.url);
+    return new ChangeFeedIterator<T>(
+      this.clientContext,
+      id,
+      path,
+      async () => {
+        return !!(await this.container.getPartitionKeyDefinition()).body;
```

Comment:
nit: !! looks a bit cryptic, can we use something like != null instead?

Reply:
!! is valid JavaScript for seeing if an object is truthy (aka not undefined or null). Agree this is confusing to read at first, though, so I refactored into two lines with some meaningful variable names